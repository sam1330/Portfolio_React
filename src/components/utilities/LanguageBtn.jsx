import React from "react";
import { useTranslation } from "react-i18next";
import Modal from "../modal/Modal";

import "./LanguageBtn.scss"

const LanguageBtn = () => {
  const [t, i18n] = useTranslation();

  return (
    <>
      <span className="language__btn">
        <svg
          xmlns="http://www.w3.org/2000/svg"
          class="icon icon-tabler icon-tabler-language"
          width="30"
          height="30"
          viewBox="0 0 24 24"
          stroke-width="2"
          stroke="currentColor"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
        >
          <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
          <path d="M4 5h7"></path>
          <path d="M9 3v2c0 4.418 -2.239 8 -5 8"></path>
          <path d="M5 9c-.003 2.144 2.952 3.908 6.7 4"></path>
          <path d="M12 20l4 -9l4 9"></path>
          <path d="M19.1 18h-6.2"></path>
        </svg>
        {/* <svg
          xmlns="http://www.w3.org/2000/svg"
          class="icon icon-tabler icon-tabler-world"
          width="30"
          height="30"
          viewBox="0 0 24 24"
          stroke-width="2"
          stroke="currentColor"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
        >
          <path stroke="none" d="M0 0h24v24H0z" fill="none"></path>
          <circle cx="12" cy="12" r="9"></circle>
          <line x1="3.6" y1="9" x2="20.4" y2="9"></line>
          <line x1="3.6" y1="15" x2="20.4" y2="15"></line>
          <path d="M11.5 3a17 17 0 0 0 0 18"></path>
          <path d="M12.5 3a17 17 0 0 1 0 18"></path>
        </svg> */}
        {/* { t("LANGUAGE") } */}
      </span>
        <Modal />
    </>
  );
};

export default LanguageBtn;