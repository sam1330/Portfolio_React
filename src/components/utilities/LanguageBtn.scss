@import url('https://fonts.googleapis.com/css2?family=Quicksand:wght@300;500&display=swap');

.language__btn {
  font-family: 'Quicksand', sans-serif;

  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 4rem;
  background-color: hsla(0, 0%, 7%, 0.027);
  padding: 0.5rem;
  border-radius: 5px;
  cursor: pointer;
}