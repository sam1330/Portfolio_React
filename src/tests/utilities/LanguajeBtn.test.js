import React from "react";
import { shallow } from "enzyme/build";
import LanguageBtn from "../../components/utilities/LanguageBtn";

describe("test on <LanguageBtn />", () => {
  const wrapper = shallow(<LanguageBtn />);

  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});