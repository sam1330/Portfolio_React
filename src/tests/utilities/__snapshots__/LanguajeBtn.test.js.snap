// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <LanguageBtn /> should render component 1`] = `
<Fragment>
  <span
    className="language__btn"
  >
    <svg
      class="icon icon-tabler icon-tabler-language"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M4 5h7"
      />
      <path
        d="M9 3v2c0 4.418 -2.239 8 -5 8"
      />
      <path
        d="M5 9c-.003 2.144 2.952 3.908 6.7 4"
      />
      <path
        d="M12 20l4 -9l4 9"
      />
      <path
        d="M19.1 18h-6.2"
      />
    </svg>
    LANGUAGE
  </span>
</Fragment>
`;
